src/braidnet_braidnode_api.erl

```diff
@@ -174,17 +174,15 @@ call_method(State, _, _, _) ->
 
 %--- Custom EPMD API -----------------------------------------------------------
 
-forward_epmd(State = #state{epmd = undefined}, Name, PortNo) ->
+forward_epmd(State = #state{epmd_client = undefined}, Name, PortNo) ->
     % Start a new erlang epm_server connection to local EPMD daemon for this node
-    % We do it here because if the node disconnect, it will disconnecte from EPMD
+    % We do it here because if the node disconnects, it will disconnect from EPMD
     % too and unregister the node.
     ?LOG_INFO("Starting extra EPMD process for ~p on port ~p linked to ~p", [Name, PortNo, self()]),
     {ok, Pid} = gen_server:start_link(erl_epmd, [], []),
-    register(node_erl_epmd, Pid),
-    register(node_connection, self()),
     {ok, _} = gen_server:call(Pid, {register, binary_to_list(Name), PortNo, inet}, infinity),
-    State#state{epmd = Pid}.
+    State#state{epmd_client = Pid}.
 
-stop_epmd(State = #state{epmd = Pid}) ->
-    gen_server:call(Pid, stop, infinity),
-    State#state{epmd = undefined}.
+stop_epmd(undefined) -> ok;
```

Comment:
Why having forward_epmd take the state, and not stop_epmd ?
I would either change forward_epmd to return a pid, or change that back to get the state. You can still do `op_epmd(State = #state{epmd_client = undefined})`.

Reply:
`stop_epmd` is only used in terminate. No point in editing a state if it is going to be destryed, I think...

Reply:
I understand, I like consistency though :)